src/checktable/mod.rs

```diff
@@ -139,6 +140,10 @@ pub struct CheckTable {
     // For each domain, stores the set of base nodes that it receives updates from.
     domain_dependencies: HashMap<domain::Index, Vec<NodeIndex>>,
 
+    // Domains impacted by each replay path.
+    replay_paths: HashMap<ReplayPath, Vec<domain::Index>>,
```

Comment:
Down the line, we want to use `VecMap` for this, `domain_dependencies`, `last_replay`, etc.